x/oracle/keeper/aggregator/aggregator.go

```diff
@@ -48,6 +48,34 @@ type aggregator struct {
 	dsPrices map[uint64]string
 }
 
+func (agg *aggregator) copy4CheckTx() *aggregator {
+	ret := &aggregator{
+		finalPrice:  big.NewInt(0).Set(agg.finalPrice),
+		reportPower: big.NewInt(0).Set(agg.reportPower),
+		totalPower:  big.NewInt(0).Set(agg.totalPower),
+
+		reports:  make([]*reportPrice, 0, len(agg.reports)),
+		dsPrices: agg.dsPrices,
```

Comment:
We may need to make a totally new map? Seemed like here referenced to the same map with original `agg.dsPrices`

Reply:
fixed